src/hotspot/share/classfile/defaultMethods.cpp

```diff
@@ -1073,7 +1058,11 @@ static void merge_in_new_methods(InstanceKlass* klass,
   Array<int>* original_ordering = klass->method_ordering();
   Array<int>* merged_ordering = Universe::the_empty_int_array();
 
-  int new_size = klass->methods()->length() + new_methods->length();
+  int new_methods_length = klass->methods()->length() + new_methods->length();
+  assert(new_methods_length <= USHRT_MAX,
+         "error methods for default method processing caused too many methods in class %s",
+         klass->external_name());
```

Comment:
I don't think this can be an assert. What happens if you define an interface with max(u2)-1 methods and have it implemented by a class that already has some methods? I would expect this to be a runtime exception.

Reply:
I expected something to check the number of miranda methods, but I don't see that.  I'll add back the exception.